File: Siteware/ClientApp/src/app/shared/models/Product.ts
import SaleTypes from "./SaleType";

class Product {
    id: number;
    name: string;
    price: number;
    saletype: SaleTypes

    constructor(
        name: string,
        price: number,
        saletype: SaleTypes
    ) {
        this.name = name;
        this.price = price;
        this.saletype= saletype;
    }
}

export default Product;

File: Siteware/Models/Product.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace Siteware.Models
{
    public class Product
    {
        public Product() {}
        public Product(int id, string name, double price, SaleTypes saleType = SaleTypes.NONE)
        {
            this.id = id;
            this.name = name;
            this.price = price;
            this.saletype = saleType;
        }

        public int id { get; set; }
        public string name { get; set; }
        public double price { get; set; }
        [EnumDataType(typeof(SaleTypes))]
        public SaleTypes saletype { get; set; }
    }
}
